proto/palomachain/paloma/gravity/genesis.proto

```diff
@@ -0,0 +1,124 @@
+// a signature for a batch or valset, or to submit a ethereum_signature for a
+// particular attestation nonce. In the case of attestations this clock starts
+// when the attestation is created, but only allows for slashing once the event
+// has passed
+//
+// target_eth_tx_timeout:
+//
+// This is the 'target' value for when ethereum transactions time out, this is a
+// target because Ethereum is a probabilistic chain and you can't say for sure
+// what the block frequency is ahead of time.
+//
+// average_block_time
+// average_ethereum_block_time
+//
+// These values are the average Cosmos block time and Ethereum block time
+// respectively and they are used to compute what the target batch timeout is.
+// It is important that governance updates these in case of any major, prolonged
+// change in the time it takes to produce a block
+//
+// slash_fraction_signer_set_tx
+// slash_fraction_batch
+// slash_fraction_ethereum_signature
+// slash_fraction_conflicting_ethereum_signature
+//
+// The slashing fractions for the various gravity related slashing conditions.
+// The first three refer to not submitting a particular message, the third for
+// submitting a different ethereum_signature for the same Ethereum event
+message Params {
+  option (gogoproto.stringer) = false;
+
+  string gravity_id = 1;
+  string contract_source_hash = 2;
+  string bridge_ethereum_address = 4;
+  uint64 bridge_chain_id = 5;
+  uint64 signed_signer_set_txs_window = 6;
+  uint64 signed_batches_window = 7;
+  uint64 ethereum_signatures_window = 8;
+  uint64 target_eth_tx_timeout = 10;
+  uint64 average_block_time = 11;
+  uint64 average_ethereum_block_time = 12;
+  // TODO: slash fraction for contract call txs too
+  bytes slash_fraction_signer_set_tx = 13 [
+    (gogoproto.customtype) = "github.com/cosmos/cosmos-sdk/types.Dec",
+    (gogoproto.nullable) = false
+  ];
+  bytes slash_fraction_batch = 14 [
+    (gogoproto.customtype) = "github.com/cosmos/cosmos-sdk/types.Dec",
+    (gogoproto.nullable) = false
+  ];
+  bytes slash_fraction_ethereum_signature = 15 [
+    (gogoproto.customtype) = "github.com/cosmos/cosmos-sdk/types.Dec",
+    (gogoproto.nullable) = false
+  ];
+  bytes slash_fraction_conflicting_ethereum_signature = 16 [
+    (gogoproto.customtype) = "github.com/cosmos/cosmos-sdk/types.Dec",
+    (gogoproto.nullable) = false
+  ];
+  uint64 unbond_slashing_signer_set_txs_window = 17;
+}
+
+// GenesisState struct
+// TODO: this need to be audited and potentially simplified using the new
+// interfaces
+message GenesisState {
+  Params params = 1;
+  uint64 last_observed_event_nonce = 2;
+  repeated google.protobuf.Any outgoing_txs = 3;
+  repeated google.protobuf.Any confirmations = 4;
+  repeated EthereumEventVoteRecord ethereum_event_vote_records = 9;
+  repeated MsgDelegateKeys delegate_keys = 10;
+  repeated ERC20ToDenom erc20_to_denoms = 11;
+  repeated SendToEthereum unbatched_send_to_ethereum_txs = 12;
```

Comment:
I suggest adding `(gogoproto.nullable) = false` to all of these, unless there's an actual difference here between empty set and a nil set.

Reply:
Same for other cases of `repeated` fields in the changed proto files.